independent-projects/tools/devtools-testing/src/main/java/io/quarkus/devtools/testing/codestarts/QuarkusCodestartTestBuilder.java

```diff
@@ -111,13 +117,16 @@ public QuarkusCodestartTestBuilder standaloneExtensionCatalog() {
                     }
                 });
                 extensions.add(extension);
-                if (quarkusVersion == null) {
-                    quarkusVersion = getBuiltWithQuarkusCore(extension);
+                if (buildWithQuarkusCoreVersion == null) {
+                    buildWithQuarkusCoreVersion = getBuiltWithQuarkusCore(extension);
                 }
             }
-            Objects.requireNonNull(quarkusVersion, "quarkus version not found in extensions");
+            Objects.requireNonNull(buildWithQuarkusCoreVersion, "quarkus version not found in extensions");
+            enableRegistryClientTestConfig(quarkusBomGroupId != null ? quarkusBomGroupId : "io.quarkus",
+                    quarkusBomVersion != null ? quarkusBomVersion : buildWithQuarkusCoreVersion);
             final ExtensionCatalog extensionCatalog = QuarkusProjectHelper
-                    .getExtensionCatalog(quarkusVersion);
+                    .getExtensionCatalog(buildWithQuarkusCoreVersion);
```

Comment:
While the BOM version and the Quarkus versions could be different, the way it's done now, it should be the same quarkusBomVersion or builtWithQuarkusCoreVersion, if quarkusBomVersion is null. To do it properly we would have to also add another argument for the quarkusVersion.

Reply:
Good catch! It's fixed